### PR TITLE
Add check-manifest to plugin_template

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #7656

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,9 @@ The following settings are stored in `template_config.yml`.
   check_commit_message  Include inspection of commit message for a reference to an issue in
                         pulp.plan.io.
 
+  check_manifest        Runs check-manifest to see if any files that were unintentionally left out
+                        MANIFEST.in. See https://pypi.org/project/check-manifest/.
+
```
Comment (fao89) [inline README.md:65]: ❤️

File: templates/bootstrap/pyproject.toml.j2 [hunk 2]
```diff
@@ -60,3 +60,20 @@ exclude = '''
 )/
 '''
 {% endif %}
+
+{% if check_manifest %}
+[tool.check-manifest]
+ignore = [
+    ".bumpversion.cfg",
+    "CHANGES/**",
```
Comment (mdellweg) [inline templates/bootstrap/pyproject.toml.j2:68]: Is this check run when doing a release? At that point, CHANGES should actually be empty.

Comment (daviddavis) [reply templates/bootstrap/pyproject.toml.j2:68]: It's run as part of our test suite so it's run against PRs and releases.

It's technically not empty at release time as there's still a `.TEMPLATE.rst` file in it.

**Review by bmbouter — APPROVED** [review]
This looks great to me. Thank you!

**Pull request merged**